spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/LocalAdressResolver.java

```diff
@@ -0,0 +1,129 @@
+package org.springframework.cloud.sleuth.util;
+
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.util.Enumeration;
+
+import org.apache.commons.logging.Log;
+
+/**
+ * Utility class for getting local Ip address from network interfaces.
+ * Picks up first non loopback and up address.
+ * @author Matcin Wielgus
+ */
+public class LocalAdressResolver {
```

Comment:
i was googling like crazy could not find, but was hoping for exactly this reaction:) I'll change it.